tests/integration/cattletest/core/test_projects.py

```diff
@@ -0,0 +1,418 @@
+
+
+def test_members_cant_delete(project_clients, members):
+    project = _create_project_with_members(project_clients['Owner'], members)
+    project = project_clients['Member'].by_id("project", project.id)
+    got_members = get_plain_members(project.projectMembers())
+    id = acc_id(project_clients['Member'])
+    for member in got_members:
+        if member['externalId'] == id:
+            assert member['role'] == 'member'
+    with pytest.raises(ApiError) as e:
+        project_clients['Member'].delete(project)
+    assert e.value.error.status == 403
+    _set_members(project_clients['Member'], project.id, [{
+        'externalId': acc_id(project_clients['Member']),
+        'externalIdType': 'rancher_id',
+        'role': 'owner'
+    }], 403)
+
+
+def test_inactive_project(project_clients, members):
+    project = _create_project(project_clients, 'Owner')
+    project_clients['admin'].wait_success(project.deactivate())
+    state = project_clients['admin'].by_id('project', project.id).state
+    assert state == 'inactive'
+    with pytest.raises(ApiError) as e:
+        project.setmembers(members=members)
+    assert e.value.error.status == 404
+    project = project_clients['admin'].by_id('project', project.id)
+    with pytest.raises(ApiError) as e:
+        project.setmembers(members=members)
+    assert e.value.error.status == 404
+
+
+def test_project_cant_create_project(project_clients, members):
+    project = _create_project(project_clients, 'Owner')
+    uuid = project.uuid
+    client = client_for_project(project)
+    with pytest.raises(ApiError) as e:
+        project = client.create_project()
+    assert e.value.error.status == 403
+    got_project = client.list_project()[0]
+    assert got_project.uuid == uuid
+    assert len(project.projectMembers()) == len(got_project.projectMembers())
+    pass
+
+
+def test_accessible_projects(project_clients):
+    members = _create_members(project_clients, ['OutThereUser'])
+    projects = []
+    for i in range(0, 4):
+        projects.append(_create_project_with_members(project_clients['admin'],
+                                                     members))
+    assert len(projects) == 4
+    got_projects = project_clients['OutThereUser'].list_project()
+    project_ids = get_ids(projects)
+    got_project_ids = get_ids(got_projects)
+    assert len(got_project_ids.intersection(project_ids)) == 4
+    for project in projects:
+        project.setmembers(members=_create_members(project_clients, ['Owner']))
+    got_projects = project_clients['OutThereUser'].list_project()
+    project_ids = get_ids(projects)
+    got_project_ids = get_ids(got_projects)
+    assert len(got_project_ids.intersection(project_ids)) == 0
+
+
+def test_create_project_no_owner(project_clients):
+    project = project_clients['admin'].create_project()
+    project = project_clients['admin'].wait_success(project)
+    PROJECTS.add(project.id)
+    assert len(project.projectMembers()) == 0
```

Comment:
I think that test is already done with all our other tests. Since when you perform an action it is done as an account and a project is account and all our other tests use accounts. If you are listed as a member you can act as the project.